src/interpreter.rs

```diff
@@ -116,10 +115,30 @@ impl Interpreter {
         })
     }
 
+    pub fn set_schedule(&mut self, schedule: Option<Schedule>) {
+        self.schedule = schedule;
+    }
+
+    pub fn set_functions(&mut self, functions: FunctionTable) {
+        self.functions = functions;
+    }
+
+    pub fn set_modules(&mut self, modules: &[Ref<Module>]) {
+        self.modules = modules.to_vec();
+    }
+
     pub fn get_modules(&mut self) -> &mut Vec<Ref<Module>> {
         &mut self.modules
     }
 
+    pub fn set_init_data(&mut self, init_data: Value) {
+        self.init_data = init_data;
+    }
+
+    pub fn get_init_data(&mut self) -> &mut Value {
```

Comment:
`get_init_data_mut` is a better name since a mutable reference is returned.

Alternatively we could have a `merge_init_data(&mut self, v : Value)` that performs a merge of init data. That way we don't need to return mutable references.

Reply:
Apparently I don't use `get_init_data` anylonger, I forgot to delete it.
There is a `set_init_data(&mut self, v : Value)` that does what you expect?